jerry-core/api/jerry.c

```diff
@@ -2956,33 +2956,104 @@ jerry_init_property_descriptor_fields (jerry_property_descriptor_t *prop_desc_p)
 {
-  jerry_assert_api_available ();
+  jerry_property_descriptor_t prop_desc;
+  jerry_init_property_descriptor_fields (&prop_desc);
 
-  if (!ecma_is_value_object (obj_val)
-      || !ecma_is_value_prop_name (prop_name_val))
+  if (prop_desc_p->flags & (ECMA_PROP_IS_ENUMERABLE_DEFINED))
   {
-    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p)));
+    prop_desc.is_enumerable_defined = true;
+    prop_desc.is_enumerable = prop_desc_p->flags & (ECMA_PROP_IS_ENUMERABLE);
   }
 
-  if ((prop_desc_p->is_writable_defined || prop_desc_p->is_value_defined)
-      && (prop_desc_p->is_get_defined || prop_desc_p->is_set_defined))
+  if (prop_desc_p->flags & (ECMA_PROP_IS_CONFIGURABLE_DEFINED))
   {
-    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p)));
+    prop_desc.is_configurable_defined = true;
+    prop_desc.is_configurable = prop_desc_p->flags & (ECMA_PROP_IS_CONFIGURABLE);
+  }
+
+  prop_desc.is_value_defined = prop_desc_p->flags & (ECMA_PROP_IS_VALUE_DEFINED);
+
+  if (prop_desc.is_value_defined)
+  {
+    if (ecma_is_value_error_reference (prop_desc_p->value))
+    {
+      prop_desc.value = ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p));
+      return prop_desc;
+    }
+
+    prop_desc.value = prop_desc_p->value;
+  }
+
+  if (prop_desc_p->flags & (ECMA_PROP_IS_WRITABLE_DEFINED))
+  {
+    prop_desc.is_writable_defined = true;
+    prop_desc.is_writable = prop_desc_p->flags & (ECMA_PROP_IS_WRITABLE);
   }
 
+  if (prop_desc_p->flags & (ECMA_PROP_IS_GET_DEFINED))
+  {
+    ecma_value_t getter = ecma_make_object_value (prop_desc_p->get_p);
+    prop_desc.is_get_defined = true;
+
+    if (ecma_is_value_error_reference (getter))
+    {
+      prop_desc.value = ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p));
+      return prop_desc;
+    }
+
+    if (ecma_op_is_callable (getter))
+    {
+      prop_desc.getter = getter;
+    }
+    else if (!ecma_is_value_null (getter))
+    {
+      prop_desc.value = ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p));
+      return prop_desc;
+    }
+  }
+
+  if (prop_desc_p->flags & (ECMA_PROP_IS_SET_DEFINED))
+  {
+    ecma_value_t setter = ecma_make_object_value (prop_desc_p->set_p);
+    prop_desc.is_set_defined = true;
+
+    if (ecma_is_value_error_reference (setter))
+    {
+      prop_desc.value = ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p));
+      return prop_desc;
+    }
+
+    if (ecma_op_is_callable (setter))
+    {
+      prop_desc.setter = setter;
+    }
+    else if (!ecma_is_value_null (setter))
+    {
+      prop_desc.value = ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p));
+      return prop_desc;
+    }
+  }
+
+  return prop_desc;
+} /* ecma_property_descriptor_to_jerry */
+
+/**
+ * Convert a jerry_property_descriptor_t to a ecma_property_descriptor_t
+ *
+ * @return raised error - if the operation fail
+ *         ecma_property_descriptor_t - if the operation is a success
```

Comment:
Ditto, always return with ecma_property_descriptor_t,
just note if any conversion error occurs, the property descriptor's `value` field is filled with `ECMA_VALUE_ERROR`